lang/funcs/core/net/cidr_to_ip.go

```diff
@@ -0,0 +1,48 @@
+// Mgmt
+// Copyright (C) 2013-2020+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package net
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/dbakong/mgmt/lang/funcs/simple"
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func init() {
+	simple.ModuleRegister(ModuleName, "cidr_to_ip", &types.FuncValue{
+		T: types.NewType("func(a str) str"),
+		V: CidrToIP,
+	})
+}
+
+// CidrToIP returns the IP from a CIDR address
+func CidrToIP(input []types.Value) (types.Value, error) {
+	cidr := input[0].Str()
+	ip, _, err := net.ParseCIDR(strings.TrimSpace(cidr))
+	if err != nil {
+		// return nil, errwrap.Wrapf(err, "invalid CIDR address: %v", cidr.Str())
```

Comment:
why is one commented out? Which is it?
Also use the correct type instead of %v when it's known. Post a link the manual for each to show you know what they all are =D

Reply:
I should've deleted the commented line. These are formatting verbs a few are below:
%v = formatting verb for value
%s = formatting verb for strings (which I should've used)

Reference: https://golang.org/pkg/fmt/